Source/WebCore/platform/graphics/cocoa/GraphicsContextGLCocoa.mm

```diff
@@ -686,9 +686,10 @@ static bool needsEAGLOnMac()
     UNUSED_VARIABLE(internalFormat);
     const EGLint attributes[] = { EGL_METAL_TEXTURE_ARRAY_SLICE_ANGLE, layer, EGL_NONE };
 #else
+    EGLint internalFormatAttr = (internalFormat) ? EGL_TEXTURE_INTERNAL_FORMAT_ANGLE : EGL_NONE;
     const EGLint attributes[] = {
         EGL_METAL_TEXTURE_ARRAY_SLICE_ANGLE, layer,
-        EGL_TEXTURE_INTERNAL_FORMAT_ANGLE, static_cast<EGLint>(internalFormat),
+        internalFormatAttr, static_cast<EGLint>(internalFormat),
         EGL_NONE
```

Comment:
Feels a bit weird to build the attributes array like this, as opposed to e.g. having a `Vector` and building it up depending on `internalFormat`.

Reply:
Fewer `malloc`s?